src/test/run-pass/for-loop-has-unit-body.rs

```diff
@@ -9,7 +9,13 @@
 // except according to those terms.
 
 fn main() {
-    for i in 0..3 {
+    // Check that the tail statement in the body unifies with something
+    for _ in 0..3 {
         unsafe { std::mem::uninitialized() }
     }
+
+    // Check that the tail statement in the body can be unit
+    for _ in 0..3 {
+        ()
```

Comment:
That's not *that* interesting, since it doesn't directly check for the property I care about, i.e. that non-`()` bodies *are denied*.